packages/@atjson/document/src/document.ts

```diff
@@ -274,7 +275,35 @@ export class Document {
     if (start < 0 || start > this.content.length)
       throw new Error("Invalid position.");
 
-    let insertion = new Insertion(start, text, behaviour);
+    let behaviourLeading;
+    let behaviourTrailing;
+    // "preserve" and "modify" are targeted for deprecation as valid options in
+    // favor of preserveLeading, preserveTrailing, or preserveBoth
+    switch (behaviour) {
```

Comment:
Yeah, this is a _little_ gross, but while we're supporting a bunch of different options, a `switch` made sense